receiver/deps.py

```
@@ -149,13 +152,29 @@ def get_config_source(db, key: str, env_map: dict, db_prefix: str) -> str:
+    Returns True if the SIGUSR2 was delivered (pkill found a matching
+    process), False if no process was found or an exception occurred.
+    Callers should log a warning on False; the config is always committed
+    to the DB before this call so a failed signal is not data loss.
+    """
     try:
-        subprocess.run(['pkill', '-SIGUSR2', '-f', '/app/main.py'],
-                      check=False, timeout=2)
+        result = subprocess.run(['pkill', '-SIGUSR2', '-f', '/app/main.py'],
+                                check=False, timeout=2)
+        signaled = result.returncode == 0
         with open('/tmp/config_update_requested', 'w') as f:
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp/config_update_requested"

(S108)